src/WorkItemMigrator/JiraExport/JiraItem.cs

```diff
@@ -47,13 +47,9 @@ private static List<JiraRevision> BuildRevisions(JiraItem jiraItem, IJiraProvide
             if (createdOn == DateTime.MinValue)
                 Logger.Log(LogLevel.Debug, "created key was not found, using DateTime default value");
 
-
-            var changelog = jiraProvider.DownloadChangelog(issueKey).ToList();
+            var changelog = jiraProvider.DownloadChangelog(issueKey).OrderByDescending(c => (long)c.SelectToken("$.id")).ToList();
             Logger.Log(LogLevel.Debug, $"Downloaded issue: {issueKey} changelog.");
 
-            if (jiraProvider.GetSettings().UsingJiraCloud)
```

Comment:
What I meant was this part that you removed. For server and cloud cases. And you are sure in both cases we get the changelog order from the API the same way?

Reply:
and line 55 *

Reply:
Ignore this